packages/clerk-js/src/ui/elements/Menu.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { PropsWithChildren, useEffect, useRef } from 'react';
+
+import { createContextAndHook } from '../../ui/utils';
+import { Button, Col } from '../customizables';
+import { usePopover } from '../hooks';
+import { animations, PropsOfComponent } from '../styledSystem';
+import { colors } from '../utils/colors';
+import { Portal } from './Portal';
+
+type UsePopoverReturn = ReturnType<typeof usePopover>;
+
+type MenuState = {
+  popoverCtx: UsePopoverReturn;
+};
+
+const [MenuStateCtx, useMenuState] = createContextAndHook<MenuState>('MenuState');
+
+type MenuProps = PropsWithChildren<Record<never, never>>;
+
+export const Menu = (props: MenuProps) => {
+  const popoverCtx = usePopover({
+    placement: 'right-start',
+    offset: 8,
+  });
+
+  return (
+    <MenuStateCtx.Provider
+      value={{ value: { popoverCtx } }}
+      {...props}
+    />
+  );
+};
+
+type MenuButtonProps = PropsOfComponent<typeof Button>;
+
+export const MenuButton = (props: MenuButtonProps) => {
+  const { popoverCtx } = useMenuState();
+  const { reference, toggle } = popoverCtx;
+
+  return (
+    <Button
+      ref={reference}
+      onClick={toggle}
+      {...props}
+    />
+  );
+};
+
+const findMenuItem = (el: Element, siblingType: 'prev' | 'next', options?: { countSelf?: boolean }) => {
```

Comment:
I'm a bit unsure about the intention of the `countSelf` param here. Can you elaborate?

Reply:
When using the function at the `MenuList`, we want the first `button` found, including the element passed to the function, so in that case, `countSelf` would be `true`.